src/block/parser.rs

```diff
@@ -278,10 +280,13 @@ impl BlockParser {
 
             log_file.read_to_end(&mut log_file_contents).await?;
 
+            let global_slot_number = extract_global_slot_number(&String::from_utf8_lossy(&log_file_contents));
```

Comment:
The String::from_utf8_lossy converts the Vec<u8> to a String, which is then passed to the extract_global_slot_number function I wrote to extract this value since unlike blockchain_length and state_hash, global_slot_number cannot be extracted from the file name.

Reply:
I meant as opposed to `from_utf8`, but I guess it’s because the log contents can have invalid utf8. Should no longer be relevant if you get this value from `protocol_state`

Reply:
It's in `consensus_state` so that may imply adding another field for consensus state, similar to protocol state, and then instead of adding another field for global_slot_since_genesis rather extract it similar to how previous_state is with a function rather than a field.

Reply:
@Isaac-DeFrain if we do not add a parameter to the function, then we have to read the file. Given this function is named `parse_file` I'm assuming it is best not to add another parameter, read the file inside the function, extract `global_slot_since_genesis` from it rather than from `consensus_state`. Moving back in this direction. Unless you think there's another way?